src/scanpy/plotting/_stacked_violin.py

```diff
@@ -823,7 +825,9 @@ def stacked_violin(
     )
 
     if dendrogram:
-        vp.add_dendrogram(dendrogram_key=dendrogram)
+        vp.add_dendrogram(
+            dendrogram_key=dendrogram if isinstance(dendrogram, str) else None
```

Comment:
The typing then of `add_dendrogram` with `dendrogram_key: str | None = None,` seems off, no? 

https://github.com/scverse/scanpy/blob/578e57297e8cfa48e5492e58b45567217a6ae43b/src/scanpy/plotting/_baseplot_class.py#L239C9-L245

Reply:
Oh apologies, I misread this.

Reply:
I think I got to the bottom of this. commit incoming